libraries/chain/block_header_state.cpp

```diff
@@ -4,6 +4,16 @@
 
 namespace eosio::chain {
 
+#warning Add last_proposed_finalizer_policy_generation to snapshot_block_header_state_v3, see header file TODO
+
+   namespace detail {
+      uint32_t get_next_next_round_block_num( block_timestamp_type t, uint32_t block_num ) {
+         auto index = t.slot % config::producer_repetitions; // current index in current round
+         //                 (increment to the end of this round  ) + next round
+         return block_num + (config::producer_repetitions - index) + config::producer_repetitions;
+      }
+   }
+
```

Comment:
I moved this in a new file in my PR so as not to duplicate, however we can checkin as-is and I'll remove in my PR.